### PR TITLE
fcitx5: pass FCITX_ADDON_DIRS to fcitx5-qt via configtool

### PR DESCRIPTION
**Pull request by linsui**
## Description of changes

fcitx5 configtool load gui addons via `fcitx5-qt5-gui-wrapper` from `fcitx5-qt`. However, `fcitx5-qt5-gui-wrapper` finds gui addons in `FCITX_ADDON_DIRS` which needs to be set to the path in `fcitx5-with-addons` so that gui addons from addons can be found. So the `FCITX_ADDON_DIRS` needs to be added to the wrapper in `with-addons.nix`. However (another however), [`fcitx5-configtool` find `fcitx5-qt` in its `buildInputs`](https://github.com/fcitx/fcitx5-configtool/blob/fd53565aab2dd908269f2cccbd0d2bfeb7f13b1c/CMakeLists.txt#L65) so that it doesn't use the wrapper of `fcitx5-qt5-gui-wrapper` in `fcitx5-with-addons`. So `fcitx5-configtool` is wrapped instead and the `FCITX_ADDON_DIRS` is inherited.

This is a proper fix for https://github.com/NixOS/nixpkgs/pull/245920.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@NickCao

### CODE REVIEW

File: pkgs/tools/inputmethods/fcitx5/with-addons.nix [hunk 1]
```diff
@@ -12,7 +20,12 @@ symlinkJoin {
       --prefix FCITX_ADDON_DIRS : "$out/lib/fcitx5" \
       --suffix XDG_DATA_DIRS : "$out/share" \
       --suffix PATH : "$out/bin" \
-      --suffix LD_LIBRARY_PATH : ${lib.makeLibraryPath (lib.flatten (map (x: x.extraLdLibraries or []) addons))}
+      --suffix LD_LIBRARY_PATH : "${lib.makeLibraryPath (lib.flatten (map (x: x.extraLdLibraries or []) addons))}"
+
+    ${lib.optionalString (fcitx5-configtool != null)''
```
Comment (NickCao) [inline pkgs/tools/inputmethods/fcitx5/with-addons.nix:25]: When would `fcitx5-configtool` be `null`?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -7818,7 +7818,9 @@ with pkgs;
 
   fcitx5 = libsForQt5.callPackage ../tools/inputmethods/fcitx5 { };
 
-  fcitx5-with-addons = libsForQt5.callPackage ../tools/inputmethods/fcitx5/with-addons.nix { };
+  fcitx5-with-addons = libsForQt5.callPackage ../tools/inputmethods/fcitx5/with-addons.nix {
+    makeWrapper = makeBinaryWrapper;
```
Comment (NickCao) [inline pkgs/top-level/all-packages.nix:7822]: Anything wrong with the default shell wrapper?

File: pkgs/tools/inputmethods/fcitx5/with-addons.nix (hunk 1 above)
Comment (linsui) [reply pkgs/tools/inputmethods/fcitx5/with-addons.nix:25]: `pkgs.fcitx5-with-addons.override { fcitx5-configtool = null; }`

I can override the package to disable the configtool.

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (linsui) [reply pkgs/top-level/all-packages.nix:7822]: It works but the I thought the C wrapper is better?

Comment (NickCao) [reply pkgs/top-level/all-packages.nix:7822]: Then just use `makeBinaryWrapper` directly, no need to override it here.

File: pkgs/tools/inputmethods/fcitx5/with-addons.nix (hunk 1 above)
Comment (NickCao) [reply pkgs/tools/inputmethods/fcitx5/with-addons.nix]: yes

**Review by NickCao — APPROVED** [review]

**Pull request merged**